solutions/cses-1757.mdx

```diff
@@ -242,3 +246,77 @@ int main() {
 	trav(t,ans) pr(t,' ');
 }
 ```
+
+</CPPSection>
+
+<JavaSection>
+```java
+import java.io.*;
+import java.util.*;
+
+public class Main {
+	public static void main(String[] args) {
+		Kattio io = new Kattio();
+		int n = io.nextInt();
+		int m = io.nextInt();
+		List<List<Integer>> graph = new ArrayList<>();
+		for (int i = 0; i <= n; i++) {
+			graph.add(new LinkedList<>());
+		}
+		for (int i = 0; i < m; i++) {
+			int a = io.nextInt();
+			int b = io.nextInt();
+			graph.get(b).add(a);
+		}
+
+		// Calculate the in degree of each node.
+		int[] inDegree = new int[n + 1];
+		for (int i = 1; i <= n; i++) {
+			for (int to : graph.get(i)) {
+				inDegree[to]++;
+			}
+		}
+
+		/*
+		 * Any node with in degree 0 can be used, so we store all possible 
+		 * nodes in a max heap to get the node with the maximum id.
+		*/
+		PriorityQueue<Integer> possibleNodes = new PriorityQueue<>(new Comparator<Integer>() {
+			@Override
+			public int compare(Integer i1, Integer i2) {
+				return -i1.compareTo(i2);
+			}
+		});
```

Comment:
think you can simplify this w/ a lambda or `Comparator.comparingInt`

Reply:
ok just fixed